syntax = "proto3";

package processeddata;

message ProcessedData {
    uint32 sequence_num = 1;
    double rx_sample_rate = 2;
    double output_sample_rate = 3;
    repeated OutputDataSet outputdataset = 4; // 1 set for every frequency
    double processing_time = 5;
    double initialization_time = 6;
    double sequence_start_time = 7;
    string rf_samples_location = 8;
    bool gps_locked = 9;
    double gps_to_system_time_diff = 10;
    uint32 agc_status_bank_h = 11;
    uint32 lp_status_bank_h = 12;
    uint32 agc_status_bank_l = 13;
    uint32 lp_status_bank_l = 14;
    string bfiq_main_shm = 15;
    string bfiq_intf_shm = 16;
    uint32 num_samps = 17;
    uint32 max_num_beams = 18;
    repeated DebugData debug_data = 19;  // 1 for each filter output stage (including antennas_iq)
    uint32 rawrf_num_samps = 20;

    message DebugData {
        string stagename = 1;
        string main_shm = 2;
        string intf_shm = 3;
        uint32 num_samps = 4;
    }

    message OutputDataSet {
        string mainacf = 1;
        string intacf = 2;
        string xcf = 3;
        uint32 num_beams = 4;
        uint32 num_ranges = 5;
        uint32 num_lags = 6;
        uint32 slice_id = 7;
        repeated float noise_at_freq = 8; //TODO fill and parse
    }
}
